lib/stat_tracker.rb

```diff
@@ -222,23 +222,28 @@ def team_goals_league_total
   end
 
   def avg_team_goals_league
-    team_games_league_total.reduce(Hash.new) do |team_avgs, (team, tot_games)|
-      team_goals_league_total.each do |team, tot_goals|
-        team_avgs[team] = (tot_goals.to_f / tot_games).round(2)
+      team_avgs = Hash.new
+      @teams.each do |team|
+        team_games = team_games_league_total[team.team_id]
+        team_goals = team_goals_league_total[team.team_id]
+        goal_stat = (team_goals.to_f / team_games.to_f).round(2)
+        team_avgs[team.team_id] = goal_stat
       end
       team_avgs
-    end
   end
 
   def max_avg_team_goals_league
-    max_avg = avg_team_goals_league.values.max
+    averages_with_nan = avg_team_goals_league.values
```

Comment:
interested in what this may have been solving; would love to talk more on call about this code

Reply:
It kinda tricked down from me using the ENTIRE teams list to calculate averages. It was getting much messier to do it based off of the games.csv

It isnt an issue with the entire data set, but once it is smaller, some teams are not part of the fixture, creating div/0 errors.